contracts/SpokePoolV3Periphery.sol

```diff
@@ -491,6 +536,14 @@ contract SpokePoolV3Periphery is SpokePoolV3PeripheryInterface, Lockable, MultiC
             s
         );
 
+        // Verify that the signatureOwner signed the input depositData.
+        if (
```

Comment:
Is this code block the same for all gasless flows? Can we refactor into internal func?

Reply:
It is the same. Factored out here: https://github.com/across-protocol/contracts/pull/790/commits/dfa6f89bcbbfc3fea759b06ed37cd600b0a64668